### PR TITLE
Add end to end testing for different application integrations

### PR DESCRIPTION
**Pull request by rbclark**
This integrates capybara and chrome into the CI pipeline and runs all the different applications on all the possible supported ruby and rails versions, validating that the applications function correctly in CI.

I spent a lot of time trying to make this work as a single Rake task, and came to the conclusion that it was basically not possible without a lot of unnecessary struggle. I've gone ahead and replaced the single rake task with 4 dedicated rake tasks:

```
bundle exec rake test_authlogic
bundle exec rake test_clearance
bundle exec rake test_devise
bundle exec rake test_sorcery
 ```

As I did in the other PR I'll add a few inline notes to discuss some of the changes.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
-test/dummy*/db/*.sqlite3-journal
-test/dummy*/log/*.log
-test/dummy*/tmp/
-test/dummy*/.sass-cache
+test/rails_apps/*/.sass-cache
+test/rails_apps/*/db/*.sqlite3
+test/rails_apps/*/db/*.sqlite3-journal
+test/rails_apps/*/log/*.log
+test/rails_apps/*/tmp/
+test/rails_apps/*/.sass-cache
```
Comment (rbclark) [inline .gitignore:9]: I ended up moving all the rails apps to be named consistently which helped clean up things in a few spots including here.

File: Rakefile [hunk 2]
```diff
@@ -14,24 +14,39 @@ RDoc::Task.new(:rdoc) do |rdoc|
   rdoc.rdoc_files.include('lib/**/*.rb')
 end
 
-APP_RAKEFILE = File.expand_path("../test/dummy/Rakefile", __FILE__)
+APP_RAKEFILE = File.expand_path("../test/rails_apps/devise/Rakefile", __FILE__)
```
Comment (rbclark) [inline Rakefile:17]: In order to maintain consistency with other spots the devise app is still the default.

File: app/controllers/any_login/application_controller.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 module AnyLogin
   class ApplicationController < ActionController::Base
     include AnyLogin.provider::Controller
+    include ::Clearance::Controller if AnyLogin.provider.to_s.eql?("AnyLogin::Provider::Clearance")
```
Comment (rbclark) [inline app/controllers/any_login/application_controller.rb:4]: This being loaded the way it was before causes problems in the test suite. This is meant to be a temporary workaround until https://github.com/igorkasyanchuk/any_login/pull/47 is merged which changes this again.

File: test/rails_apps/authlogic/config/environments/test.rb [hunk 4]
```diff
@@ -14,7 +14,7 @@
   # Eager loading loads your whole application. When running a single test locally,
   # this probably isn't necessary. It's a good idea to do in a continuous integration
   # system, or in some way before deploying your code.
-  config.eager_load = ENV["CI"].present?
+  config.eager_load = false
```
Comment (rbclark) [inline test/rails_apps/authlogic/config/environments/test.rb:17]: I spent a bunch of time trying to figure out why this was happening and wasn't able to get to the bottom of it. Just disabling eager loading for authlogic seemed to be the easiest workaround. Just for background, enabling `eager_load` causes problems with authlogic itself loading properly.

**Comment by igorkasyanchuk** [comment]
@rbclark I want to merge it :) but it has conflicts
thank you again for your tremendous work

**Comment by rbclark** [comment]
@igorkasyanchuk Sorry this should be all set now!

File: test/rails_apps/sorcery/db/migrate/20150909210357_add_role_to_users.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class AddRoleToUsers < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :role, :string
+  end
+
```
Comment (manastyretskyi) [inline test/rails_apps/sorcery/db/migrate/20150909210357_add_role_to_users.rb:5]: I think you can remove this line

**Comment by igorkasyanchuk** [comment]
merged 👍

**Pull request merged**